.github/workflows/comment-download.yml

```
@@ -0,0 +1,100 @@
+name: Comment on pull request
+on:
+  workflow_run:
+    workflows: ['Documentation', 'Build Python wheels']
+    types: [completed]
+
+permissions:
+    pull-requests: write
+
+jobs:
+  pr_comment:
+    if: github.event.workflow_run.event == 'pull_request' && github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v7
+        with:
+          script: |
+            async function getArtifactLink(artifactName, owner, repo, run_id) {
+                // get the list of artifacts
+                const artifacts = await github.paginate(
+                    github.rest.actions.listWorkflowRunArtifacts, {owner, repo, run_id}
+                );
+
+                if (!artifacts.length) {
+                    return core.error(`No artifacts found`);
+                }
+
+                for (const artifact of artifacts) {
+                    if (artifact.name == artifactName) {
+                        return `https://nightly.link/${owner}/${repo}/actions/artifacts/${artifact.id}.zip`;
+                    }
+                }
+                return core.error(`failed to find ${artifactName} artifact`);
+            }
+            let link;
+            let body_message;
+
+            const {owner, repo} = context.repo;
+            const run_id = ${{github.event.workflow_run.id}};
+            const workflow_name = "${{github.event.workflow_run.name}}";
+    
+            console.info(`${workflow_name} triggered this run`)
+
+            // Find the PR with the right sha, see https://github.com/orgs/community/discussions/25220
+            const response = await github.rest.search.issuesAndPullRequests({
+                q: 'repo:${{ github.repository }} is:pr sha:${{ github.event.workflow_run.head_sha }}',
+                per_page: 1,
+            })
+
+            const items = response.data.items
+            if (items.length < 1) {
+                return core.error("No matching pull requests found");
+            }
+
+            const pull_number = items[0].number;
+            console.info("Pull request number is", pull_number)
+
+            if (workflow_name == 'Documentation') {
+                link = await getArtifactLink('docs', owner, repo, run_id);
+                // Add horizontal line for docs only because assuming this is one
+                // triggered first. The wheels building should take much longer...
+                body_message = `\n----\n 📖 [Download documentation](${link})\n`;
+            } else { // wheels
+                link = await getArtifactLink('wheels', owner, repo, run_id);
+                body_message = `⚙️ [Download wheels](${link})\n`;
```

```suggestion
                body_message = `⚙️ [Download Python wheels for this pull-request (you can install these with `pip`)](${link})\n`;
```

or something similar